collector/logical_router_port_collector_test.go

```diff
@@ -0,0 +1,189 @@
+package collector
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/go-vmware-nsxt/manager"
+)
+
+const (
+	fakeLogicalRouterPortID          = "fake-logical-router-port-id"
+	fakeLogicalRouterPortDisplayName = "fake-logical-router-port-name"
+	fakeLogicalRouterID              = "fake-logical-router-id"
+)
+
+type mockLogicalRouterPortClient struct {
+	responses                  []mockLogicalRouterPortResponse
+	logicalRouterPortListError error
+}
+
+type mockLogicalRouterPortResponse struct {
+	ID              string
+	DisplayName     string
+	LogicalRouterID string
+	Error           error
+
+	RxTotalBytes     int64
+	RxTotalPackets   int64
+	RxDroppedPackets int64
+	TxTotalBytes     int64
+	TxTotalPackets   int64
+	TxDroppedPackets int64
+}
+
+func (c *mockLogicalRouterPortClient) ListAllLogicalRouterPorts() ([]manager.LogicalRouterPort, error) {
+	if c.logicalRouterPortListError != nil {
+		return nil, c.logicalRouterPortListError
+	}
+	var logicalRouterPorts []manager.LogicalRouterPort
+	for _, response := range c.responses {
+		logicalRouterPort := manager.LogicalRouterPort{
+			Id:              response.ID,
+			DisplayName:     response.DisplayName,
+			LogicalRouterId: response.LogicalRouterID,
+		}
+		logicalRouterPorts = append(logicalRouterPorts, logicalRouterPort)
+	}
+	return logicalRouterPorts, nil
+}
+
+func (c *mockLogicalRouterPortClient) GetLogicalRouterPortStatisticsSummary(lrportID string) (manager.LogicalRouterPortStatisticsSummary, error) {
+	for _, res := range c.responses {
+		if res.ID == lrportID {
+			return manager.LogicalRouterPortStatisticsSummary{
+				Rx: &manager.LogicalRouterPortCounters{
+					TotalBytes:     res.RxTotalBytes,
+					TotalPackets:   res.RxTotalPackets,
+					DroppedPackets: res.RxDroppedPackets,
+				},
+				Tx: &manager.LogicalRouterPortCounters{
+					TotalBytes:     res.TxTotalBytes,
+					TotalPackets:   res.TxTotalPackets,
+					DroppedPackets: res.TxDroppedPackets,
+				},
+			}, res.Error
+		}
+	}
+	return manager.LogicalRouterPortStatisticsSummary{}, errors.New("error")
+}
+
+func buildLogicalRouterPortResponse(id string, baseValue int64, err error) mockLogicalRouterPortResponse {
+	return mockLogicalRouterPortResponse{
+		ID:               fmt.Sprintf("%s-%s", fakeLogicalRouterPortID, id),
+		DisplayName:      fmt.Sprintf("%s-%s", fakeLogicalRouterPortDisplayName, id),
+		LogicalRouterID:  fmt.Sprintf("%s-%s", fakeLogicalRouterID, id),
+		Error:            err,
+		RxTotalBytes:     baseValue * 3,
```

Comment:
line 80-85: any value in making this dynamic? why not just use constant value

Reply:
if it's using constant across all responses, the implementation might cheat?
But I agree using multiplication here is weird, removing the multiplication